### PR TITLE
linux: replace the initramfs source during post_patch

### PR DESCRIPTION
**Pull request by lrusak**
This makes it so we don't have any surprise options appearing or disappearing after the kernel has been unpacked.

Maintainers please update your configs:
 - [x] Amlogic @chewitt 
 - [x] Rockchip @Kwiboo 
 - [x] RPi/Generic @MilhouseVH 
 - [x] Allwinner @jernejsk 

aarch64 kernels may have initramfs compression (if the kernel is uncompressed)
arm kernels do not use initramfs compression (the entire kernel is compressed)

We should probably unify that at some point but it is outside of the scope of this PR.

I can either pick your commits and squash them in or you can push them directly to this branch (or you can PR to my branch).

Thanks.

### CODE REVIEW

**Comment by MilhouseVH** [comment]
@lrusak thanks for this. Please squash the RPi/Generic updates using https://github.com/LibreELEC/LibreELEC.tv/commit/f1ffbd7d86794ceefae913486a4f1c1bb69db5cc

There's another couple of "annoyances" when running `make oldconfig` that are capable of introducing build host specific changes:
```
# Compiler: gcc (Ubuntu 7.2.0-8ubuntu3.2) 7.2.0
```
and
```
CONFIG_GCC_VERSION=70200
```
Both of these changes should ideally be ignored when submitting a config change to avoid flip-flopping these options/comments whenever one developer submits a minor option change while using a different version of `gcc` to the previous config submitter. It's a cosmetic thing, but just adds potential noise to a config change and maybe something to bear in mind (as it's not specifically called out in the workflow notes).

**Comment by MilhouseVH** [comment]
Allwinner: https://github.com/LibreELEC/LibreELEC.tv/commit/f2665c28760a78b8d335d43dd4f10e3da534fa29

Amlogic: https://github.com/LibreELEC/LibreELEC.tv/commit/13175aa753a64899e221ef15d5e7258d1e4dc012

Rockchip: https://github.com/LibreELEC/LibreELEC.tv/commit/3ba8b2601e3bcec64fb8d7852465eafefdf49b06 (plus extras, @Kwiboo please double check)

**Comment by lrusak** [comment]
@MilhouseVH thanks. I've left the commits separate right now until I get a sign off from each maintainer :+1:

File: packages/linux/package.mk [hunk 1]
```diff
@@ -66,8 +66,7 @@ fi
 post_patch() {
   cp $PKG_KERNEL_CFG_FILE $PKG_BUILD/.config
 
-  sed -i -e "s|^CONFIG_INITRAMFS_SOURCE=.*$|CONFIG_INITRAMFS_SOURCE=\"$BUILD/image/initramfs.cpio\"|" $PKG_BUILD/.config
-  sed -i -e '/^CONFIG_INITRAMFS_SOURCE=*./ a CONFIG_INITRAMFS_ROOT_UID=0\nCONFIG_INITRAMFS_ROOT_GID=0' $PKG_BUILD/.config
+  sed -i -e "s|@INITRAMFS_SOURCE@|\"$BUILD/image/initramfs.cpio\"|" $PKG_BUILD/.config
```
Comment (MilhouseVH) [inline packages/linux/package.mk]:
```suggestion
  sed -e "s|@INITRAMFS_SOURCE@|\"$BUILD/image/initramfs.cpio\"|" -i $PKG_BUILD/.config
```

Comment (MilhouseVH) [reply packages/linux/package.mk]: More complete fixup here: https://github.com/LibreELEC/LibreELEC.tv/commit/d38983d45c7d753321e92ce12b57f04f78bbb34f

**Comment by MilhouseVH** [comment]
Ping @chewitt :smile:

**Comment by chewitt** [comment]
fine with me, I'll make the changes (or adjust to whatever was merged) when I next update the current draft PR .. will be a while due to work and other real-life things.

**Pull request merged**